library-app/src/components/Layout/MainLayout.tsx

```diff
@@ -1,32 +1,64 @@
 import { useEffect, useState } from 'react'
+import { Route, Routes } from 'react-router-dom'
+import { BookBodyDataGet } from '../../models/bookData.model'
+import Where from '../../models/where.model'
+import PrivateRoutes from '../../router/PrivateRoutes'
 import Footer from '../Footer/Footer'
 import Header from '../Header/Header'
+import BookDetails from '../pages/BookDetails/BooksDetails'
+import Homepage from '../pages/Homepage/Homepage'
+import Login from '../pages/Login/Login'
 import Sidebar from '../Sidebar/Sidebar'
 import styles from './MainLayout.module.css'
 
-export interface IMainLayout {
-  children: React.ReactNode
-}
-export interface MainLayoutProps {
-  isLoggedIn: boolean
-}
+const MainLayout = () => {
+  const [books, setBooks] = useState<BookBodyDataGet[]>([])
+  const [isLoggedIn, setIsLoggedIn] = useState(true)
+  const [accessToken, setAccessToken] = useState(localStorage.getItem('accessToken'))
```

Comment:
Don't access localStorage directly from component

Reply:
Also, do you need both isLoggedIn and accessToken? You use accessToken to check if user is logged in anyway

Reply:
I could done it with just accessToken, but i thought it would be easier to have one state named isLoggedIn also. But later i realized that only one state would be easier to cope with